File: python/darts/darts.py
def score(x, y):
    # calculate the distance (squared) from the centre
    dist = x**2 + y**2

    # determine how many points were earned
    if dist > 100:  # 10^2
        return 0
    elif dist > 25: # 5^2
        return 1
    elif dist > 1:
        return 5
    else:
        return 10

File: java/darts/src/main/java/Darts.java
class Darts {

    private double x;
    private double y;

    Darts(double x, double y) {
        this.x = x;
        this.y = y;
    }

    int score() {

        int pts = 0;
        double dist = Math.sqrt(this.x * this.x + this.y * this.y);
        if ( dist <= 1.0 ) {
            pts = 10;
        } else if ( dist <= 5.0 ) {
            pts = 5;
        } else if ( dist <= 10.0 ) {
            pts = 1;
        }

        return pts;
    }

}
